### PR TITLE
Respond to query if no nodes available

### PR DESCRIPTION
**Pull request by mrferris**
### What was wrong?
Queries were timing out when no nodes were available for the query. This was always the case, but having no nodes available  became something that started happening when #802 was merged, which only considers connected nodes.

### How was it fixed?

1. Use closest nodes regardless of connectivity if all nodes are in a disconnected state
2. If no nodes whatsoever are in our routing table, send a "no content found" response immediately instead of never responding. Currently change 1 makes it so that this should never happen, but better safe than sorry.

### To-Do

[//]: # (Stay ahead of things, add list items here!)
- [ ] Add entry to the [release notes](https://github.com/ethereum/trin/blob/master/newsfragments/README.md) (may forgo for trivial changes)
- [ ] Clean up commit history

### CODE REVIEW

**Comment by mrferris** [comment]
Note that the unit test that I added fails on master with a `canceled` error (like in #821 ), and it passes with this fix.

**Review by KolbyML — APPROVED** [review]
This is nice :+1:

File: portalnet/src/overlay_service.rs
Comment (carver) [inline portalnet/src/overlay_service.rs:2410]: Hm, I was wondering why this needs to be a write lock when it seems to be a read-only query. I'm assuming you tried a read lock first. Assuming you did, I think the `&mut self` on `closest_values` is why it's required. Though I can't tell from looking at the source why it couldn't be a `&self` instead:
https://github.com/sigp/discv5/blob/1439decd4e7d7c9de78ef61b5d67be3fee688510/src/kbucket.rs#L650-L670

Might be worth seeing if it's as easy as removing the `mut` in a draft PR against discv5. (To be clear, I *don't* mean that getting the discv5 change should block this PR).

File: portalnet/src/overlay_service.rs
Comment (carver) [inline portalnet/src/overlay_service.rs:2407]: This is an concerning enough situation that it seems worth a `warn!` log (or maybe at least `info!`). Having 0 known connected nodes is noteworthy.

**Review by carver — APPROVED** [review]
Yup, looks great. Yay for red/green testing!

File: portalnet/src/overlay_service.rs
Comment (mrferris) [reply portalnet/src/overlay_service.rs:2411]: I wondered the same. 

I'll see if that discv5 change breaks anything after merging.

**Pull request merged**
